server/belga/macros/set_default_metadata_with_translate.py

```diff
@@ -79,29 +79,22 @@ def set_default_metadata_with_translate(item, **kwargs):
         return
 
     # SDBELGA-538
-    if item.get("subject"):
-        filtered_subjects = [
-            s for s in item["subject"] if s.get("scheme") == "services-products"
-        ]
-        internal_destination = kwargs.get("internal_destination", {})
-        content_filter_id = internal_destination.get("filter", "")
-
-        # Retrieve the content filter using content_filter_id
-        content_filter = get_resource_service("content_filters").find_one(
-            req=None, _id=content_filter_id
-        )
-
-        if len(filtered_subjects) > 1 and content_filter:
-            filter_conditions_service = get_resource_service("filter_conditions")
-
-            for filter in content_filter.get("content_filter", []):
-                for filter_id in filter.get("expression", {}).get("fc", []):
-                    filter_condition = filter_conditions_service.find_one(
-                        req=None, _id=filter_id
-                    )
-
-                    if filtered_subjects[:1] != filter_condition.get("value"):
-                        raise StopDuplication
+    filtered_subjects = [
+        s
+        for s in item["subject"]
+        if item.get("subject") and s.get("scheme") == "services-products"
+    ]
+    internal_destination = kwargs.get("internal_destination", {})
+    content_filter_id = internal_destination.get("filter", "")
+    content_filter_service = get_resource_service("content_filters")
+    content_filter = content_filter_service.find_one(req=None, _id=content_filter_id)
+
+    if (
+        len(filtered_subjects) > 1
+        and content_filter
+        and content_filter_service.does_match(content_filter, item)
```

Comment:
you have to call it with item that has only that single service, something like:
```python3
test_item = item.copy()
test_item["subject"] = [s for s in item["subject"] if s.get("scheme") == "services-products"][:1]  # only first if there is one
...
content_filter_service.does_match(content_filter, test_item)
```

Reply:
I tried this condition but this will not stop it, because `content_filter_service.does_match(content_filter, test_item)` returns **true**